spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -158,11 +163,18 @@ public SubscriptionAdminClient subscriptionAdminClient() {
 					SubscriptionAdminSettings.newBuilder()
 							.setCredentialsProvider(this.finalCredentialsProvider)
 							.setHeaderProvider(this.headerProvider)
+							.setTransportChannelProvider(transportChannelProvider())
 							.build());
 		}
 		catch (IOException ioe) {
-			throw new PubSubException("An error occurred while creating SubscriptionAdminClient.",
-					ioe);
+			throw new PubSubException("An error occurred while creating SubscriptionAdminClient.", ioe);
 		}
 	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	@ConditionalOnExpression("'${spring.cloud.gcp.pubsub.emulatorHost:}' == ''")
```

Comment:
Since `GcpPubSubEmulatorConfiguration` is loaded before this `GcpPubSubAutoConfiguration` the condition like this is redundant. And I somehow think it is a mix of concerns to mention this property exactly in this class.
If it is unavoidable, then there is something wrong and we should spot it and fix.